pkg/controller/linstorcontrollerset/linstorcontrollerset_controller.go

```diff
@@ -302,81 +302,132 @@ func (r *ReconcileLinstorControllerSet) reconcileControllers(ctx context.Context
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if len(ourPods.Items) > 1 {
+		log.WithField("#controllerPods", len(ourPods.Items)).Debug("requeue because multiple controller pods are present")
+		return &reconcileutil.TemporaryError{
+			RequeueAfter: time.Minute,
+			Source: fmt.Errorf("multiple controller pods present"),
+		}
+	}
+
 	return nil
 }
 
-func (r *ReconcileLinstorControllerSet) reconcileControllerNodeWithControllers(ctx context.Context, pcs *piraeusv1alpha1.LinstorControllerSet, pod corev1.Pod) error {
+func (r *ReconcileLinstorControllerSet) reconcileStatus(ctx context.Context, pcs *piraeusv1alpha1.LinstorControllerSet, resErr error) error {
 	log := logrus.WithFields(logrus.Fields{
-		"podName":      pod.Name,
-		"podNameSpace": pod.Namespace,
-		"podPhase":     pod.Status.Phase,
+		"Name":      pcs.Name,
+		"Namespace": pcs.Namespace,
 	})
-	log.Debug("CS reconcileControllerNodeWithControllers: reconciling node")
+	log.Info("reconcile status")
 
-	log.Debug("check if controller is reachable")
-
-	_, err := r.linstorClient.Nodes.GetControllerVersion(ctx)
+	log.Debug("find active controller pod")
+	pod, err := r.findActiveControllerPod(ctx, pcs)
 	if err != nil {
-		return &reconcileutil.TemporaryError{
-			Source:       err,
-			RequeueAfter: connectionRetrySeconds * time.Second,
-		}
+		log.Warnf("failed to find active controller pod: %v", err)
 	}
 
-	log.Debug("register controller")
+	controllerName := ""
+	if pod != nil {
+		controllerName = pod.Name
+	}
 
-	ctrl := pcs.Status.ControllerStatus
-	ctrl.NodeName = pod.Spec.NodeName
+	pcs.Status.ControllerStatus = &piraeusv1alpha1.NodeStatus{
+		NodeName: controllerName,
+		RegisteredOnController: false,
+	}
 
-	// Mark this true on successful exit from this function.
-	ctrl.RegisteredOnController = false
+	log.Debug("check if controller pod is registered")
 
-	node, err := r.linstorClient.GetNodeOrCreate(ctx, lapi.Node{
-		Name: pod.Name,
-		Type: lc.Controller,
-		NetInterfaces: []lapi.NetInterface{
-			{
-				Name:                    "default",
-				Address:                 pod.Status.PodIP,
-				SatellitePort:           pcs.Spec.SslConfig.Port(),
-				SatelliteEncryptionType: pcs.Spec.SslConfig.Type(),
-			},
-		},
-	})
+	allNodes, err := r.linstorClient.Nodes.GetAll(ctx)
 	if err != nil {
-		return err
+		log.Warnf("failed to fetch list of LINSTOR nodes: %v", err)
 	}
 
-	log.WithFields(logrus.Fields{
-		"nodeName": node.Name,
-		"nodeType": node.Type,
-	}).Debug("CS reconcileControllerNodeWithControllers: found node")
+	for _, node := range allNodes {
+		if pod != nil && node.Name == pod.Name {
+			pcs.Status.ControllerStatus.RegisteredOnController = true
+		}
+	}
+
+	log.Debug("fetch information about storage nodes")
 
 	nodes, err := r.linstorClient.GetAllStorageNodes(ctx)
 	if err != nil {
-		return fmt.Errorf("CS unable to get cluster storage nodes: %v", err)
+		log.Warnf("unable to get LINSTOR storage nodes: %v", err)
```

Comment:
If `err != nil`, I wouldn't trust the other values returned. Maybe set `nodes` to an empty slice explicitly here?

Reply:
nice catch!